src/Runner/FileFilterIterator.php

```diff
@@ -54,15 +57,6 @@ public function __construct(
     public function accept(): bool
     {
         $file = $this->current();
-        if (!$file instanceof \SplFileInfo) {
```

Comment:
tbh I'm not sure. one can ignore param annotation and pass whatever iterator they want.
until PHP starts support templated types on language level, i would keep the check here.

Reply:
You can't check the content of every iterable... I'd say until PHP starts supporting templated types on the language level, we need to rely on static analysis :)

Reply:
I would agree and keep the check for public APIs where we have no guarantee downstream users will use static analysis, but this class is internal so static analysis is sufficient.

Reply:
Actually, it receives a finder from the user config with no type check in between, and `ConfigInterface::setFinder()` does not specify iterable type, so downstream user can pass any iterable easily and wouldn't even break the API.

Reply:
Right, good catch!

Reply:
You can use https://phpstan.org/config-reference#treatphpdoctypesascertain in the phpstan config, if you want to still be able to do check like this